src/main/java/teammates/common/datatransfer/attributes/StudentAttributes.java

```diff
@@ -19,75 +21,78 @@
+        return builder(student.getCourseId(), student.getName(), student.getEmail())
+                .withLastName(student.getLastName())
+                .withComments(student.getComments())
+                .withTeam(student.getTeamName())
+                .withSection(student.getSectionName())
+                .withGoogleId(student.getGoogleId())
+                .withKey(student.getRegistrationKey())
+                .withCreatedAt(student.getCreatedAt())
+                .withUpdatedAt(student.getUpdatedAt())
+                .build();
     }
 
-    public StudentAttributes(String section, String team, String name, String email, String comment,
-                             String courseId) {
-        this();
-        this.section = section;
-        this.team = team;
-        this.lastName = SanitizationHelper.sanitizeName(StringHelper.splitName(name)[1]);
-        this.name = SanitizationHelper.sanitizeName(name);
-        this.email = email;
-        this.comments = SanitizationHelper.sanitizeTextField(comment);
-        this.course = courseId;
-    }
-
-    public StudentAttributes(CourseStudent student) {
-        this();
-        this.email = student.getEmail();
-        this.course = student.getCourseId();
-        this.name = student.getName();
-        this.lastName = student.getLastName();
-        this.comments = SanitizationHelper.sanitizeTextField(student.getComments());
-        this.team = student.getTeamName();
-        this.section = student.getSectionName() == null ? Const.DEFAULT_SECTION : student.getSectionName();
-        this.googleId = student.getGoogleId() == null ? "" : student.getGoogleId();
-        this.key = student.getRegistrationKey();
-
-        this.createdAt = student.getCreatedAt();
-        this.updatedAt = student.getUpdatedAt();
-
-    }
-
-    private StudentAttributes(StudentAttributes other) {
-        this(other.googleId, other.email, other.name, other.comments,
-             other.course, other.team, other.section);
-        this.key = other.key;
-        this.updateStatus = other.updateStatus;
+    /**
+     * Return new builder instance with default values for optional fields.
+     *
+     * <p>Following default values are set to corresponding attributes:
+     * <ul>
+     * <li>{@code googleId = ""}</li>
+     * <li>{@code section = Const.DEFAULT_SECTION}</li>
+     * <li>{@code updateStatus = StudentUpdateStatus.UNKNOWN}</li>
+     * <li>{@code createdAt = Const.TIME_REPRESENTS_DEFAULT_TIMESTAMP}</li>
+     * <li>{@code updatedAt = Const.TIME_REPRESENTS_DEFAULT_TIMESTAMP}</li>
+     * <li>{@code lastName = SanitizationHelper.sanitizeName(StringHelper.splitName(name)[1])}</li>
+     * </ul>
+     */
+    public static Builder builder(String courseId, String name, String email) {
+        return new Builder(courseId, name, email);
     }
 
     public StudentAttributes getCopy() {
-        return new StudentAttributes(this);
+        return builder(course, name, email)
```

Comment:
Similarly, will `this(toEntity())` work?

Reply:
It won't work in this case because StudentAttributesEnitity doesn't contain updateStatus field. Also, there no setter for registrationKey field - it's generated inside entity so we cannot be sure that this field will be same as in StudentAttributes.

Reply:
In that case, will delegating most of the work to `valueOf(toEntity())` and setting the rest manually work?
```java
StudentAttributes copy = valueOf(toEntity());
copy.* = *; // as necessary
return copy;
```